internal/indexer/processors/effects.go

```diff
@@ -0,0 +1,255 @@
+		"auth_clawback_enabled_flag",
+	}
+	trustlineFlags = []string{
+		"authorized_flag",
+		"authorized_to_maintain_liabilites",
+		"clawback_enabled_flag",
+	}
+)
+
+type EffectsProcessor struct {
+	networkPassphrase string
+}
+
+func NewEffectsProcessor(networkPassphrase string) *EffectsProcessor {
+	return &EffectsProcessor{
+		networkPassphrase: networkPassphrase,
+	}
+}
+
+func (p *EffectsProcessor) ProcessTransaction(ctx context.Context, tx ingest.LedgerTransaction, op xdr.Operation, opIdx uint32) ([]types.StateChange, error) {
+	ledgerCloseTime := tx.Ledger.LedgerCloseTime()
+	ledgerNumber := tx.Ledger.LedgerSequence()
+	opID := toid.New(int32(ledgerNumber), int32(tx.Index), int32(opIdx)).ToInt64()
+	txID := toid.New(int32(ledgerNumber), int32(tx.Index), 0).ToInt64()
+
+	opWrapper := operation.TransactionOperationWrapper{
+		Index:          opIdx,
+		LedgerSequence: tx.Ledger.LedgerSequence(),
+		LedgerClosed:   time.Unix(tx.Ledger.LedgerCloseTime(), 0),
+		Transaction:    tx,
+		Operation:      op,
+		Network:        p.networkPassphrase,
+	}
+
+	effectOutputs, err := effects.Effects(&opWrapper)
+	if err != nil {
+		return nil, fmt.Errorf("processing effects: %w", err)
+	}
+
+	stateChanges := make([]types.StateChange, 0)
+	masterBuilder := NewStateChangeBuilder(ledgerNumber, ledgerCloseTime, txID).WithOperationID(opID)
+	for _, effect := range effectOutputs {
+		changeBuilder := masterBuilder.Clone().WithAccount(effect.Address)
+
+		effectType := effects.EffectType(effect.Type)
+		//exhaustive:ignore
+		switch effectType {
+		case effects.EffectSignerCreated, effects.EffectSignerRemoved, effects.EffectSignerUpdated:
+			signerWeight := effect.Details["weight"]
+			signerPublicKey := effect.Details["public_key"]
+			stateChanges = append(stateChanges, changeBuilder.
+				WithCategory(types.StateChangeCategorySigner).
+				WithReason(signerEffectToReasonMap[effect.Type]).
+				WithSigner(signerPublicKey.(string), signerWeight).
+				Build())
+
+		case effects.EffectAccountThresholdsUpdated:
+			changeBuilder = changeBuilder.WithCategory(types.StateChangeCategorySignatureThreshold)
+			stateChanges = append(stateChanges, p.parseThresholds(changeBuilder, &effect)...)
+
+		case effects.EffectAccountFlagsUpdated:
+			changeBuilder = changeBuilder.WithCategory(types.StateChangeCategoryFlags)
+			stateChanges = append(stateChanges, p.parseFlags(accountFlags, changeBuilder, &effect)...)
+
+		case effects.EffectAccountHomeDomainUpdated:
+			keyValueMap := p.parseKeyValue([]string{"home_domain"}, &effect)
+			stateChanges = append(stateChanges, changeBuilder.
+				WithCategory(types.StateChangeCategoryMetadata).
+				WithReason(types.StateChangeReasonHomeDomain).
+				WithKeyValue(keyValueMap).
+				Build())
+
+		case effects.EffectTrustlineFlagsUpdated:
+			changeBuilder = changeBuilder.WithCategory(types.StateChangeCategoryTrustlineFlags)
+			stateChanges = append(stateChanges, p.parseFlags(trustlineFlags, changeBuilder, &effect)...)
```

Comment:
In the old design doc we have this category named `BalanceAuthorization`, because you can be deauthorized to hold a contract token, and contract tokens won't have trustlines. I'm not sure when it got changed but maybe I missed more discussion?